src/utils/create-utils.ts

```diff
@@ -151,3 +158,53 @@ export const createComponentDetectionQuery = async (
 
   return cdq.status.componentDetected;
 };
+
+/**
+ * Create SPIAccessTokenBinding CR
+ *
+ * @param url the URL of the git repository
+ * @param namespace namespace to create the binding
+ * @returns Returns created SPIAccessTokenBinding resource
+ *
+ * TODO: Return type any should be changed to a proper type like K8sResourceCommon
+ */
+export const initiateAccessTokenBinding = async (url: string, namespace: string) => {
+  const name = uniqueId(`hac-dev-import-${uid()}`);
+  const requestData = {
+    apiVersion: `${SPIAccessTokenBindingModel.apiGroup}/${SPIAccessTokenBindingModel.apiVersion}`,
+    kind: SPIAccessTokenBindingModel.kind,
+    metadata: {
+      name,
+      namespace,
+    },
+    spec: {
+      repoUrl: url,
+      permissions: {
+        required: [
+          { type: 'r', area: 'repository' },
+          { type: 'w', area: 'repository' },
+        ],
+      },
+      secret: {
+        name: 'hac-dev-repo-token',
+        type: 'kubernetes.io/basic-auth',
+      },
+    },
+  };
+
+  let binding: SPIAccessTokenBindingKind = await k8sCreateResource({
+    model: SPIAccessTokenBindingModel,
+    data: requestData,
+  });
+
+  while (!binding.status) {
+    await new Promise((r) => setTimeout(r, 200));
+    binding = await k8sGetResource({
```

Comment:
What's the point polling here? Seems odd when we should be using a watch.

Reply:
Updated to use watch.